### ISSUE TITLE
cdk deploy: fails with cryptic error if notices request times out

### ISSUE THREAD

**Issue by shahinism** [aws/aws-cdk#26431]
### Describe the bug

When I run `cdk deploy`, it fails with the following error and exit code 1:

```
Subprocess exited with error null
```

No extra information is printed out. However, when I use `--verbose` it raises the following info:

```
[13:31:21] outdir: cdk.out
[13:31:21] env: {
  CDK_DEFAULT_REGION: 'us-east-1',
  CDK_DEFAULT_ACCOUNT: 'XXXXXXXXXXX',
  CDK_OUTDIR: 'cdk.out',
  CDK_CLI_ASM_VERSION: '32.0.0',
  CDK_CLI_VERSION: '2.87.0'
}
[13:31:24] Could not refresh notices: Error: Request timed out
Subprocess exited with error null
[13:32:38] Could not refresh notices: Error: Request timed out
[13:32:38] Error: Subprocess exited with error null
    at ChildProcess.<anonymous> (/usr/local/lib/node_modules/aws-cdk/lib/index.js:414:50537)
    at ChildProcess.emit (node:events:513:28)
    at ChildProcess.emit (node:domain:489:12)
```

The same stack is able to deployed to other accounts, but for some unknown reason it fails with some others. 

### Expected Behavior

Deploy normally, or show a more helpful error message. 

### Current Behavior

Fails with a not helpful message.

### Reproduction Steps

I'm not sure how you can reproduce this, without having this stack. 

### CDK CLI Version

2.87.0

### Node.js Version

18.16.1

### OS

Amazon Linux 2

### Language

Python

### Language Version

3.9.0

**Comment by peterwoodworth** [comment]
Does `--notices false` unblock?

**Comment by shahinism** [comment]
No, the error persists, also when `--notices false` is enabled. The following is the full command I run now:

```sh
cdk deploy --verbose --notices false --require-approval never --all
```

**Comment by shahinism** [comment]
Well, we managed to resolve this issue by increasing the memory capacity of our CI/CD worker instance from an AWS `t2-medium` to `t2-large`. It seems like it was an out-of-memory issue; however, I'm wondering:
1. Why CDK needs more than 4GB of RAM to function.
2. Why is such an error not caught before the process is killed?

**Comment by nl-brett-stime** [comment]
Try adding an entry in your package.json for the latest version of `aws-crt` .

**Comment by petedowl** [comment]
I resolved by increasing Node.js heap size

`export NODE_OPTIONS=--max-old-space-size=4096
`
